xline/src/storage/execute_error.rs

```diff
@@ -1,134 +1,124 @@
-        Self::AuthError(format!("role {rolename} not found"))
-    }
-
+    #[error("role {0} not found")]
+    RoleNotFound(String),
     /// Role already exists
-    pub(crate) fn role_already_exists(rolename: &str) -> Self {
-        Self::AuthError(format!("role {rolename} already exists"))
-    }
-
+    #[error("role {0} already exists")]
+    RoleAlreadyExists(String),
     /// Role not granted
-    pub(crate) fn role_not_granted(rolename: &str) -> Self {
-        Self::AuthError(format!("role {rolename} is not granted to the user"))
-    }
+    #[error("role {0} is not granted to the user")]
+    RoleNotGranted(String),
     /// Root role not exist
-    pub(crate) fn root_role_not_exist() -> Self {
-        Self::AuthError("root user does not have root role".to_owned())
-    }
-
+    #[error("root user does not have root role")]
+    RootRoleNotExist,
     /// Permission not granted
-    pub(crate) fn permission_not_granted() -> Self {
-        Self::AuthError("permission not granted to the role".to_owned())
-    }
-
+    #[error("permission not granted to the role")]
+    PermissionNotGranted,
     /// Permission not given
-    pub(crate) fn permission_not_given() -> Self {
-        Self::AuthError("permission not given".to_owned())
-    }
-
+    #[error("permission not given")]
+    PermissionNotGiven,
     /// Invalid auth management
-    pub(crate) fn invalid_auth_management() -> Self {
-        Self::AuthError("invalid auth management".to_owned())
-    }
-
+    #[error("invalid auth management")]
+    InvalidAuthManagement,
     /// Invalid auth token
-    pub(crate) fn invalid_auth_token() -> Self {
-        Self::AuthError("invalid auth token".to_owned())
-    }
-
+    #[error("invalid auth token")]
+    InvalidAuthToken,
     /// Token manager is not initialized
-    pub(crate) fn token_manager_not_init() -> Self {
-        Self::AuthError("token manager is not initialized".to_owned())
-    }
-
+    #[error("token manager is not initialized")]
+    TokenManagerNotInit,
     /// Token is not provided
-    pub(crate) fn token_not_provided() -> Self {
-        Self::AuthError("token is not provided".to_owned())
-    }
-
+    #[error("token is not provided")]
+    TokenNotProvided,
     /// Token is expired
-    pub(crate) fn token_old_revision(claim_rev: i64, cur_rev: i64) -> Self {
-        Self::AuthError(format!(
-            "token's revision {claim_rev} is older than current revision {cur_rev}"
-        ))
+    #[error("token's revision {0} is older than current revision {1}")]
+    TokenOldRevision(i64, i64),
+
+    /// Db error
+    #[error("db error: {0}")]
+    DbError(String),
+
+    /// Permission denied Error
+    #[error("permission denied")]
+    PermissionDenied,
+}
+
+impl From<ExecuteError> for tonic::Status {
+    #[inline]
+    fn from(err: ExecuteError) -> Self {
+        let code = match err {
+            ExecuteError::InvalidRequest(_)
+            | ExecuteError::KeyNotFound
```

Comment:
The `tonic::Code::NotFound` indicates that some requested entity was not found. Therefore, the `KeyNotFound` should be converted to tonic::CodeNotFound. FYI: https://docs.rs/tonic/0.9.2/tonic/enum.Code.html